### PR TITLE
Fix multiplication result converted to larger type reports in tests

### PR DESCRIPTION
**Pull request by dalg24**
Fix 
https://github.com/kokkos/kokkos/security/code-scanning/23
https://github.com/kokkos/kokkos/security/code-scanning/22
https://github.com/kokkos/kokkos/security/code-scanning/19
https://github.com/kokkos/kokkos/security/code-scanning/18
https://github.com/kokkos/kokkos/security/code-scanning/9
https://github.com/kokkos/kokkos/security/code-scanning/8
https://github.com/kokkos/kokkos/security/code-scanning/7
https://github.com/kokkos/kokkos/security/code-scanning/6
https://github.com/kokkos/kokkos/security/code-scanning/5
https://github.com/kokkos/kokkos/security/code-scanning/4
https://github.com/kokkos/kokkos/security/code-scanning/3

After this, only benchmarks and performance tests will be left

### CODE REVIEW

File: containers/unit_tests/TestBitset.hpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ struct TestBitset {
 
   TestBitset(bitset_type const& bitset) : m_bitset(bitset) {}
 
-  unsigned testit(unsigned collisions) {
+  unsigned testit(unsigned long long collisions) {
```
Comment (dalg24) [inline containers/unit_tests/TestBitset.hpp:42]: Warning emitted from https://github.com/kokkos/kokkos/blob/89899c18de8a1709e91e858578585774717ce6b1/containers/unit_tests/TestBitset.hpp#L46

File: algorithms/unit_tests/TestStdAlgorithmsTeamCount.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ void test_A(const bool searched_value_exist, std::size_t numTeams,
 
   using rand_pool =
       Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace>;
-  rand_pool pool(lowerBound * upperBound);
+  rand_pool pool(static_cast<uint64_t>(lowerBound) * upperBound);
```
Comment (dalg24) [inline algorithms/unit_tests/TestStdAlgorithmsTeamCount.cpp:114]: `Random_XorShift64_Pool` is constructed from `uint64_t` seed argument

File: algorithms/unit_tests/TestBinSortA.hpp [hunk 3]
```diff
@@ -77,7 +77,7 @@ struct sum3D {
 };
 
 template <class ExecutionSpace, typename KeyType>
-void test_3D_sort_impl(unsigned int n) {
+void test_3D_sort_impl(size_t n) {
   using KeyViewType = Kokkos::View<KeyType* [3], ExecutionSpace>;
 
   KeyViewType keys("Keys", n * n * n);
```
Comment (dalg24) [inline algorithms/unit_tests/TestBinSortA.hpp:83]: This is the only use of `n` in that function

File: containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp [hunk 4]
```diff
@@ -48,7 +48,7 @@ struct TestViewCtorProp_EmbeddedDim {
     void operator()(const int i) const { v(i) = i; }
   };
 
-  static void test_vcpt(const int N0, const int N1) {
+  static void test_vcpt(const size_t N0, const size_t N1) {
```
Comment (dalg24) [inline containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp:51]: N0 and N1 are used as extent arguments when constructing views, and they are occurrences where N0*N1 is passed.

File: containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp [hunk 5]
```diff
@@ -81,7 +81,7 @@ struct TestViewCtorProp_EmbeddedDim {
         ASSERT_EQ((std::is_same_v<CommonViewValueType, double>), true);
 #if 0
       // debug output
-      for ( int i = 0; i < N0*N1; ++i ) {
+      for ( size_t i = 0; i < N0*N1; ++i ) {
         printf(" Output check: hcv1(%d) = %lf\n ", i, hcv1(i) );
```
[the remark this answers is not in the document]
Comment (dalg24) [reply containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp:85]: shouldn't it be `%llu`?

[the remark this answers is not in the document]
Comment (dalg24) [reply containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp:85]: or `%zu`

[the remark this answers is not in the document]
Comment (masterleinad) [reply containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp:85]: Probably `%zu` but I would prefer casting or using `cout`.

[the remark this answers is not in the document]
Comment (dalg24) [reply containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp:85]:
```suggestion
        printf(" Output check: hcv1(%zu) = %lf\n ", i, hcv1(i) );
```

[the remark this answers is not in the document]
Comment (dalg24) [reply containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp]: Fixed.  Keep in mind we are arguing about a commented block of code (see the `#if 0` L82)

[the remark this answers is not in the document]
Comment (masterleinad) [reply containers/unit_tests/TestViewCtorPropEmbeddedDim.hpp]: Ah, I didn't notice it was commented.

**Review by masterleinad — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**